pocs/filterwheel/simulator.py

```diff
@@ -0,0 +1,119 @@
+import math
+import time
+import random
+import threading
+
+from astropy import units as u
+
+from pocs.utils import error
+from pocs.filterwheel import AbstractFilterWheel
+
+
+class FilterWheel(AbstractFilterWheel):
+    """
+    Class for simulated filter wheels.
+
+    Args:
+        name (str, optional): name of the filter wheel
+        model (str, optional): model of the filter wheel
+        camera (pocs.camera.*.Camera, optional): camera that this filter wheel is associated with.
+        filter_names (list of str): names of the filters installed at each filter wheel position
+        timeout (u.Quantity, optional): maximum time to wait for a move to complete. Should be
+            a Quantity with time units. If a numeric type without units is given seconds will be
+            assumed. Default is 10 seconds.
+        serial_number (str): serial number of the filter wheel
+        move_time (astropy.units.Quantity, optional): time to move the filter wheel by one position,
+            optional, default 1 second.
+        move_bidirectional (bool, optional): if True will simulate filter wheel which can rotate in
+            either direction, if False (default) will similate a filter wheel that only moves in
```

Comment:
`simulate`

Also, is this only a feature of the simulator? I didn't see anything about this in the actual code.

Reply:
Yes, it's used to determine how long a simulated move should take, based on whether the filter wheel can move directly to towards the requested position or whether it has to go the long way around because it only rotates in one direction.

Outside of the simulator this is an implementation detail of the filter wheel itself, so the POCS code doesn't need to know about it.

Filter wheels that only rotate in one direction are pretty common, as it improves repeatability. The SBIG filter wheel that I've been testing with is like that, but the only reason I could tell was by watching it.